src/arrays/arraymath.jl

```diff
@@ -42,69 +46,96 @@ end
 
 # operator * operator -> operator
 function *{B<:OrthonormalBasis}(qm::QuMatrix{B}, dm::DualMatrix{B})
-    return QuArray(A_mul_Bc(rawcoeffs(qm), rawcoeffs(dm)),
-                   bases(qm,1),
-                   bases(dm,2))
+    mc = A_mul_Bc(rawcoeffs(dm), rawcoeffs(qm))
+    QAT = similar_type(typeof(qm))
```

Comment:
You are right. This is actually a left-over because I had initially changed `QuMatrix` to `AbstractQuMatrix`, but this collided with the other functions. I think this has to be sorted out when we have a more general `CTranspose`. However, I should probably change it back for now.